pkg/k8shandler/service.go

```diff
@@ -93,34 +98,38 @@ func (er *ElasticsearchRequest) createOrUpdateService(serviceName, namespace, cl
 	cluster.AddOwnerRefTo(service)
 
 	err := client.Create(context.TODO(), service)
-	if err != nil {
-		if !errors.IsAlreadyExists(err) {
-			return fmt.Errorf("Failure constructing %v service: %v", service.Name, err)
-		}
+	if err == nil {
+		return nil
+	}
+	if !apierrors.IsAlreadyExists(err) {
+		return kverrors.Wrap(err, "failed to construct service",
+			"service_name", service.Name)
+	}
 
-		current := service.DeepCopy()
-		retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
-			if err = client.Get(context.TODO(), types.NamespacedName{Name: current.Name, Namespace: current.Namespace}, current); err != nil {
-				if errors.IsNotFound(err) {
-					// the object doesn't exist -- it was likely culled
-					// recreate it on the next time through if necessary
-					return nil
-				}
-				return fmt.Errorf("Failed to get %v service: %v", service.Name, err)
+	current := new(v1.Service)
```

Comment:
Why does this line switch from `DeepCopy()` to `new()`?

Reply:
Yes `&v1.Service{}` is more intuitive, but OTH it is a nit.

Reply:
They both do the exact same thing so I'm leaving it as-is for now. I don't see any reason for us to be consistent between `new(T)` and `&T{}` since they don't have any overall impact on the codebase or readability.